src/azure-cli/azure/cli/command_modules/redis/custom.py

```diff
@@ -37,12 +37,18 @@ def cli_redis_update(cmd, instance, sku=None, vm_size=None):
 
     # avoid setting memory configs for basic sku
     if instance.sku.name == 'Basic':
-        if 'maxmemory-reserved' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxmemory-reserved')
-        if 'maxfragmentationmemory-reserved' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxfragmentationmemory-reserved')
-        if 'maxmemory-delta' in instance.redis_configuration:
-            instance.redis_configuration.pop('maxmemory-delta')
+        memory_configs = ['maxmemory-reserved', 'maxfragmentationmemory-reserved', 'maxmemory-delta']
+        for memory_config in memory_configs:
+            if memory_config in instance.redis_configuration:
```

Comment:
Are there any tests that cover this change?

Reply:
No. Existing tests doesn't cover this. To test this scenario, there needs be an existing redis cache instance. Redis cache creation takes a while and as per the earlier guidelines, test wasn't allowed wait for longer duration.

Reply:
So how can we ensure your change is correct if no test covers this change?

---
In reply to: [357367878](https://github.com/Azure/azure-cli/pull/11502#discussion_r357367878) [](ancestors = 357367878)

Reply:
As mentioned earlier, there needs to be an existing cache to add test coverage for this change. Please provide guidance on how to use sleep in tests until redis cache instance becomes available. Currently, I have tested this change locally.